client/packages/system/src/IndicatorsDemographics/DetailView/IndicatorsDemographics.tsx

```diff
@@ -209,7 +207,10 @@ const IndicatorsDemographicsComponent = () => {
 
   return (
     <>
-      <AppBarButtons patch={setter} rows={Object.values(draft)}></AppBarButtons>
+      <AppBarButtons
+        addRow={(newRow: Row) => setDraft({ ...draft, [newRow.id]: newRow })}
```

Comment:
a slightly shortcutted version is possible here..
```suggestion
        addRow={newRow => setDraft({ ...draft, [newRow.id]: newRow })}
```